src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -86,27 +86,28 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig, gdDAO: Datap
   }
 
   /* Create a google bucket and populate it with init files */
-  private[service] def initializeBucket(googleProject: GoogleProject, clusterName: String, bucketName: String): Future[Unit] = {
+  private[service] def initializeBucket(googleProject: GoogleProject, clusterName: String, bucketName: String, clusterRequest: ClusterRequest): Future[Unit] = {
     for {
       _ <- gdDAO.createBucket(googleProject, bucketName)
-      _ <- initializeBucketObjects(googleProject, clusterName, bucketName)
+      _ <- initializeBucketObjects(googleProject, clusterName, bucketName, clusterRequest)
     } yield { }
   }
 
   /* Process the templated cluster init script and put all initialization files in the init bucket */
-  private[service] def initializeBucketObjects(googleProject: GoogleProject, clusterName: String, bucketName: String): Future[Unit] = {
+  private[service] def initializeBucketObjects(googleProject: GoogleProject, clusterName: String, bucketName: String, clusterRequest: ClusterRequest): Future[Unit] = {
     val initScriptPath = dataprocConfig.configFolderPath + dataprocConfig.initActionsScriptName
-    val replacements = ClusterInitValues(googleProject, clusterName, bucketName, dataprocConfig).toJson.asJsObject.fields
-
-    template(initScriptPath, replacements).map { content =>
-      val initScriptStorageObject = gdDAO.uploadToBucket(googleProject, bucketName, dataprocConfig.initActionsScriptName, content)
-      // Create an array of all the certs, cluster docker compose file, and the site.conf for the apache proxy
-      val certs = List(dataprocConfig.jupyterServerCrtName, dataprocConfig.jupyterServerKeyName, dataprocConfig.jupyterRootCaPemName,
-        dataprocConfig.clusterDockerComposeName, dataprocConfig.jupyterProxySiteConfName)
-      // Put the rest of the initialization files in the init bucket concurrently
-      val storageObjects = certs.map { certName => gdDAO.uploadToBucket(googleProject, bucketName, certName, new File(dataprocConfig.configFolderPath, certName)) }
-      // Return an array of all the Storage Objects
-    }
+    val replacements = ClusterInitValues(googleProject, clusterName, bucketName, dataprocConfig, clusterRequest).toJson.asJsObject.fields
+    val filesToUpload = List(dataprocConfig.jupyterServerCrtName, dataprocConfig.jupyterServerKeyName, dataprocConfig.jupyterRootCaPemName,
+      dataprocConfig.clusterDockerComposeName, dataprocConfig.jupyterProxySiteConfName, dataprocConfig.jupyterInstallExtensionScript)
+
+    for {
```

Comment:
Fixes a bug causing a race condition when creating clusters. Previously we were not waiting on the `uploadToBucket` futures, so the cluster creation could continue before the buckets were populated, potentially causing errors. This changes the `map` to a `flatMap`, ensuring the tasks run in the right order.